### PR TITLE
Extract fleet metadata/coreOS version and use them as tags

### PR DESCRIPTION
**Pull request by jvassev**

### CODE REVIEW

File: dd-agent.service [hunk 1]
```diff
@@ -4,19 +4,23 @@ After=docker.service
...
 ExecStartPre=-/usr/bin/docker rm dd-agent
-ExecStart=/usr/bin/bash -c \
-"/usr/bin/docker run --name dd-agent -h `hostname` \
--v /var/run/docker.sock:/var/run/docker.sock \
--v /proc/mounts:/host/proc/mounts:ro \
--v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
--e API_KEY=`etcdctl get /datadog/apikey` \
-datadog/docker-dd-agent"
+ExecStart=/bin/bash -c " \
+  metadata=$(fleetctl list-machines -fields machine,metadata -full -no-legend | grep %m | awk '{print $2}' | sed 's/=/:/g'); \
```
Comment (alq666) [inline dd-agent.service]: A simple update to handle nicely the case where metadata is `-`.

```bash
... | grep %m | awk '$2 !~ /-/ { gsub(/=/, ":"); print "," $2;}'
```

File: dd-agent.service [hunk 2]
```diff
@@ -4,19 +4,23 @@ After=docker.service
...
+  docker run --name dd-agent \
+    -h %H \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v /proc/mounts:/host/proc/mounts:ro \
+    -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
+    -e API_KEY=`etcdctl get /datadog/apikey` \
+    -e TAGS=\"os-name:$NAME,os-version:$VERSION,machine-id:%m,$metadata\" \
```
Comment (alq666) [inline dd-agent.service]: And here (a bit ugly I agree)

```
 -e TAGS=\"os-name:$NAME,os-version:$VERSION,machine-id:%m$metadata\"
```

**Comment by alq666** [comment]
@jvassev if the suggested updates work for you, in your environment, happy to merge when you give me the go-ahead.

File: dd-agent.service [hunk 3]
```diff
@@ -4,19 +4,23 @@ After=docker.service
...
+    datadog/docker-dd-agent;"
```
Comment (remh) [inline dd-agent.service]: We should specify a tag (likely "latest")

**Comment by jvassev** [comment]
Just added some quotes in the awk command

**Comment by remh** [comment]
@hkaj can you test and merge please ?

**Comment by hkaj** [comment]
@jvassev thanks for the contribution! Tested today and it worked well, merging.

**Pull request merged**